addons/mrp_subcontracting_dropshipping/models/purchase.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, _
+from odoo.exceptions import UserError
+
+
+class PurchaseOrder(models.Model):
```

Comment:
These purchase/dropshipping changes again seems separate from the mass SN assigning and should be in a separate commit. so it's clear what it's purpose is.

Reply:
commits splitted